server/endpoint/endpoint.go

```diff
@@ -0,0 +1,25 @@
+package endpoint
+
+import (
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/app-operator/server/middleware"
+	"github.com/giantswarm/app-operator/service"
+)
+
+// Config represents the configuration used to construct an endpoint.
+type Config struct {
+	// Dependencies
+	Logger     micrologger.Logger
+	Middleware *middleware.Middleware
+	Service    *service.Service
+}
+
+// New creates a new endpoint with given configuration.
+func New(config Config) (*Endpoint, error) {
+	return &Endpoint{}, nil
+}
+
+// Endpoint is the endpoint collection.
+type Endpoint struct {
+}
```

Comment:
FYI we try to place the structure before the constructor in general. Just flipping `New` and `Endpoint` would be nice. See also https://github.com/giantswarm/fmt/blob/master/go/struct_initialization.md.